wasm-chain-integration/src/v1/trie/low_level.rs

```diff
@@ -1087,12 +1097,14 @@ impl<V> MutableTrie<V> {
     pub fn normalize(&mut self, root: u32) {
         let new_len = root as usize + 1;
         let one_past_new_root = self.generation_roots.get(new_len).copied();
+        self.iterator_roots.push(std::collections::BTreeMap::new());
```

Comment:
This seems very wrong here. We are truncating here, so why push?

In any case, this is exactly the reason why iterator_roots should be part of generation_roots.